### PR TITLE
Add mseccfg and *envcfg CSRs

### PR DESCRIPTION
**Pull request by aswaterman**

### CODE REVIEW

**Comment by jhauser-us** [comment]
The new text has:
> If U-mode is not supported, FIOM is hardwired to zero.

If U mode isn't implemented, I would say `menvcfg` doesn't exist.  This register's mandate is to control "certain characteristics of the execution environment for modes less privileged than M".

I have no other complaints at this time.

**Comment by aswaterman** [comment]
s/complaints/comments/ right? :P

Anyway, you’re absolutely right.

**Comment by scottj97** [comment]
Does `senvcfg` affect VU-mode also? I can imagine a virtualized guest OS in VS-mode setting it.

**Comment by jhauser-us** [comment]
> Does senvcfg affect VU-mode also?

Yes.  See the hypervisor extension chapter, Section 5.2, "Hypervisor and Virtual Supervisor CSRs":
> Some standard supervisor CSRs (...) have no matching VS CSR. These supervisor CSRs continue to have their usual function and accessibility even when V=1, except with VS-mode and VU-mode substituting for HS-mode and U-mode.

**Comment by jhauser-us** [comment]
If S mode isn't implemented (but U mode is), we should allow bit FIOM to be hardwired to zero.

**Comment by aswaterman** [comment]
Ack.

**Comment by jhauser-us** [comment]
We should probably also allow FIOM to be hardwired to zero if S mode is implemented but only Bare address translation is supported.

**Comment by aswaterman** [comment]
Indeed.

**Comment by aswaterman** [comment]
I made that remark for both M and S modes, which I think is the right thing to do.

**Comment by jhauser-us** [comment]
FWIW, I think this pull request is ready to merge.

**Comment by aswaterman** [comment]
If only to reduce the complexity of juggling multiple outstanding pull requests, I'm going to merge it now.  As usual, that's not an indication that further criticism is unwelcome.

File: src/priv-csrs.tex [hunk 1]
```diff
@@ -197,6 +197,10 @@ \section{CSR Listing}
 \tt 0x105 & SRW  &\tt stvec      & Supervisor trap handler base address. \\
 \tt 0x106 & SRW  &\tt scounteren & Supervisor counter enable. \\
 \hline
+\multicolumn{4}{|c|}{Supervisor Configuration} \\
+\hline
+\tt 0x10A & MRW  &\tt senvcfg    & Supervisor environment configuration register. \\
```
Comment (scottj97) [inline src/priv-csrs.tex:202]: Shouldn't the privilege here be SRW, not MRW?

File: src/priv-csrs.tex [hunk 2]
```diff
@@ -243,6 +247,11 @@ \section{CSR Listing}
 \tt 0x64A & HRW  &\tt htinst     & Hypervisor trap instruction (transformed). \\
 \tt 0xE12 & HRO  &\tt hgeip      & Hypervisor guest external interrupt pending. \\
 \hline
+\multicolumn{4}{|c|}{Hypervisor Configuration} \\
+\hline
+\tt 0x60A & MRW  &\tt henvcfg    & Hypervisor environment configuration register. \\
+\tt 0x61A & MRW  &\tt henvcfgh   & Additional hypervisor env. conf. register, RV32 only. \\
```
Comment (scottj97) [inline src/priv-csrs.tex:253]: Shouldn't these two be HRW, not MRW?

Comment (aswaterman) [reply src/priv-csrs.tex:253]: Yes, copy-paste screwup.

**Pull request merged**